### DISCUSSION TITLE
Tacotron2 No test sentences output because of "Waveform is not finite everywhere. Skipping the GL"

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [coqui-ai/TTS#1206] [General Q&A]
Hi,

I am fine-tuning the Tacotron 2 French model (trained until 320k steps) following the official [tutorial](https://tts.readthedocs.io/en/latest/finetuning.html). But even after 835 epochs (global step 335150), the test sentence cannot be generated. So I can't hear how the model evolves. 
The only sentences it has ever output were right at the beginning (step 320216).

Yet the test sentence is not long compared to what was originally in the config.json. It is just "Nous nous sommes battus pour sa libération".

Should I increase the "max_decoder_steps" even further ? Should I set the language of the dataset (it was not set in the original config.json) ?

The log output reads as follows : 

`
> EVALUATION 

 >[!] Waveform is not finite everywhere. Skipping the GL.
WARNING:root:NaN or Inf found in input tensor.
WARNING:root:NaN or Inf found in input tensor.
...
 | > Synthesizing test sentences.
   > Decoder stopped with `max_decoder_steps` 1000
 [!] Waveform is not finite everywhere. Skipping the GL.

  --> EVAL PERFORMANCE
     | > avg_loader_time: 0.07054 (+0.00631)
     | > avg_decoder_loss: nan (+nan)
     | > avg_postnet_loss: nan (+nan)
     | > avg_stopnet_loss: nan (+nan)
     | > avg_decoder_coarse_loss: nan (+nan)
     | > avg_decoder_ddc_loss: nan (+nan)
     | > avg_ga_loss: nan (+nan)
     | > avg_decoder_diff_spec_loss: nan (+nan)
     | > avg_postnet_diff_spec_loss: nan (+nan)
     | > avg_decoder_ssim_loss: nan (+nan)
     | > avg_postnet_ssim_loss: nan (+nan)
     | > avg_loss: nan (+nan)
     | > avg_align_error: nan (+nan)

 > Number of output frames: 1

 > EPOCH: 837/1000
 --> /content/drive/MyDrive/YourTTS/Models/finetuning taco2 fr-February-05-2022_07+47PM-f45717f7

 > DataLoader initialization
 | > Use phonemes: True
   | > phoneme language: fr-fr
 | > Number of instances : 569
 | > Max length sequence: 169
 | > Min length sequence: 27
 | > Avg length sequence: 91.37609841827768
 | > Num. instances discarded by max-min (max=500000, min=6) seq limits: 0
 | >
`

Thanks

**Comment by lexkoro** [comment]
All your eval losses are NaN, there seems something wrong with your training. Is it the same for the training losses?

**Reply by Ca-ressemble-a-du-fake** [reply]
No training losses were not NaN, they were decreasing.

**Reply by TejaswiniiB** [reply]
Hi,

I too am facing the same issue. Training losses are not nan, but evaluations losses are nan. How to overcome?

**Reply by erogol** [reply]
Try disabling mixed percision training.
